src/MLJInterface.jl

```diff
@@ -0,0 +1,231 @@
+    )
+    return (;
+        best_idx=best_idx,
+        equations=formatted.trees,
+        equation_strings=equation_strings,
+        losses=formatted.losses,
+        complexities=formatted.complexities,
+        scores=formatted.scores,
+    )
+end
+
+# TODO: Pass `variable_names` and `units`
+# TODO: Enable `verbosity` being passed to `equation_search`
+MMI.clean!(::AbstractSRRegressor) = ""
+
+function MMI.fit(m::AbstractSRRegressor, verbosity, X, y, w=nothing)
+    return _update(m, verbosity, (; state=nothing), nothing, X, y, w)
+end
+function MMI.update(
+    m::AbstractSRRegressor, verbosity, old_fitresult, old_cache, X, y, w=nothing
+)
+    return _update(m, verbosity, old_fitresult, old_cache, X, y, w)
+end
+function _update(m::AbstractSRRegressor, verbosity, old_fitresult, old_cache, X, y, w)
+    options = get(old_fitresult, :options, get_options(m))
+    mX = MMI.matrix(X)
+    variable_names = getcolnames(X)
+    X_t = permutedims(mX)
+    y_t = format_input_for(m, y)
+    search_state = equation_search(
+        X_t,
+        y_t;
+        niterations=m.niterations,
+        weights=w,
+        variable_names=variable_names,
+        options=options,
+        parallelism=m.parallelism,
+        numprocs=m.numprocs,
+        procs=m.procs,
+        addprocs_function=m.addprocs_function,
+        runtests=m.runtests,
+        saved_state=old_fitresult.state,
+        return_state=true,
+        loss_type=m.loss_type,
+    )
+    fitresult = (; state=search_state, options=options)
+    return (fitresult, nothing, full_report(m, fitresult))
+end
+
+function getcolnames(X)
+    try
+        return getcolnames(MMI.schema(X), X)
+    catch e
+        isa(e, ArgumentError) || rethrow(e)
+    end
+    try
+        return getcolnames(MMI.schema(MMI.table(X)), X)
+    catch e
+        isa(e, MethodError) || isa(e, ArgumentError) || rethrow(e)
+    end
+    return getcolnames(nothing, X)
+end
+getcolnames(::Nothing, X) = [map(i -> "x$(i)", axes(X, 2))...]
+getcolnames(sch, _) = [string.(sch.names)...]
+
+format_input_for(::SRRegressor, y) = (@assert ndims(y) == 1; y)
+format_input_for(::MultitargetSRRegressor, y) = MMI.matrix(y, transpose=true)
+function MMI.fitted_params(m::AbstractSRRegressor, fitresult)
+    report = full_report(m, fitresult)
+    return (;
+        best_idx=report.best_idx,
+        equations=report.equations,
+        equation_strings=report.equation_strings,
+    )
+end
+function MMI.predict(m::SRRegressor, fitresult, Xnew)
+    params = MMI.fitted_params(m, fitresult)
+    Xnew_t = transpose(MMI.matrix(Xnew))
+    eq = params.equations[params.best_idx]
+    out, flag = eval_tree_array(eq, Xnew_t, fitresult.options)
+    !flag && error("Detected a NaN in evaluating expression.")
+    return out
+end
+function MMI.predict(m::MultitargetSRRegressor, fitresult, Xnew)
+    params = MMI.fitted_params(m, fitresult)
+    Xnew_t = transpose(MMI.matrix(Xnew))
```

Comment:
```suggestion
    Xnew_t = MMI.matrix(Xnew, transpose=true)
```

I believe you intended to change this also

Reply:
Yes thanks